ktlint-ruleset-standard/src/main/kotlin/com/pinterest/ktlint/ruleset/standard/rules/NoEmptyFileRule.kt

```diff
@@ -0,0 +1,59 @@
+package com.pinterest.ktlint.ruleset.standard.rules
+
+import com.pinterest.ktlint.rule.engine.core.api.ElementType
+import com.pinterest.ktlint.rule.engine.core.api.Rule
+import com.pinterest.ktlint.rule.engine.core.api.editorconfig.EditorConfig
+import com.pinterest.ktlint.rule.engine.core.api.editorconfig.EditorConfigProperty
+import com.pinterest.ktlint.ruleset.standard.StandardRule
+import org.ec4j.core.model.PropertyType
+import org.jetbrains.kotlin.com.intellij.lang.ASTNode
+
+public class NoEmptyFileRule :
+    StandardRule(id = "no-empty-file", usesEditorConfigProperties = setOf(NO_EMPTY_FILE_PROPERTY)),
+    Rule.Experimental {
+    private var noEmptyFile = NO_EMPTY_FILE_PROPERTY.defaultValue
+
+    override fun beforeFirstNode(editorConfig: EditorConfig) {
+        noEmptyFile = editorConfig[NO_EMPTY_FILE_PROPERTY]
+    }
+
+    override fun beforeVisitChildNodes(
+        node: ASTNode,
+        autoCorrect: Boolean,
+        emit: (offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit,
+    ) {
+        if (noEmptyFile) {
+            node
+                .takeIf { it.elementType == ElementType.FILE }
+                ?.takeIf { it.textLength == TEXT_LENGTH_EMPTY_FILE_CONTAINS }
```

Comment:
The file contains a code element other than package or import statements.
I may not understand exactly what you mean by this, so could you tell me what cases you have?

Reply:
The original issue states:
> After some refactorings we ended up with an empty file.
> 
> It would be great if there was a rule that a completely empty kt file was an error as well as a file with only a package declaration:
> 
> ```kotlin
> package my.package
> ```

So a file like:
```
package my.package

import foo.bar.Foobar

```
should be considered an empty file although the length of the content is greater than 0.

Another edge case would be a file that only contains toplevel declarations having a `private` modifier. But I am fine, with leaving that out of scope.